### DISCUSSION TITLE
get required fields before submit

### DISCUSSION THREAD

**Discussion by EvgenBabenko** [react-hook-form/react-hook-form#2220] [Q&A] [answered]
Hi guys, i'm litlle bit stuck ....
how can i get required fields and don't validate form
i want to show * if field is required, but before submitting, 
[smth like this](https://www.google.com/search?q=required+field&rlz=1C1SQJL_ruUA824UA824&sxsrf=ALeKk00YXMDdMW2T7imaWWTICeHJS5CO7g:1594741217915&source=lnms&tbm=isch&sa=X&ved=2ahUKEwj4pbX-ic3qAhWttYsKHcaPBNwQ_AUoAXoECA0QAw&biw=1536&bih=775#imgrc=r_SLVE6-NnNdbM)

thanks in advance

**Comment by bluebill1049** [comment, marked as answer]
That's a style related question right?

**Reply by EvgenBabenko** [reply]
i'm not sure
i can show this * with pass props required, but in this case when i press submit,  will work html5 validation + with yup validation and this give strange results, so that's not my case
also i can pass custom props, for example isRequired, but before that i need know which fields are required
errors object give me this info about required field, but errors works after submit

**Reply by EvgenBabenko** [reply]
and define in 2 places which fields are required is terrible (1 place it's yup schema, 2 field + required prop ), so I would like to do it in one place and pass down required fields from useForm to my components

sorry i haven't gave you information about pcks. I use v5 + yup schema

thanks in advance

**Reply by bluebill1049** [reply]
provide a simple codesandbox we can take a look.

**Reply by EvgenBabenko** [reply]
https://codesandbox.io/s/fervent-archimedes-4176j?file=/src/App.jsx

i see this *, but after press submit you can see how html5 validation work, and this validation prevent yup validation
so in this way i have defined in 2 places required fields (1 - yup schema, 2  - TextField + required prop)
if i delete this required props from TextField, yup validation works how expected, but i need know which fields are required before submit to pass throw required fields to TextField to customize TextField with styles

thanks

**Reply by bluebill1049** [reply]
https://github.com/react-hook-form/react-hook-form/discussions/2266

I think you meant `noValidate`

**Reply by bluebill1049** [reply]
please don't spam questions... I just release it's the same person.

**Reply by EvgenBabenko** [reply]
sorry, noValidate works, but in this case i have to describe in 2 places required fields, this is way full of mistakes

i think , i've found my solution:
import { yupResolver } from "@hookform/resolvers";
yupResolver(schema)().then(values => console.log(values))

so this values like errors from useForm and give me the same information, so now i can customize my TextField with style to show this *

anyway thanks!
